packages/contracts-periphery/contracts/universal/DrippieRedd.sol

```diff
@@ -0,0 +1,226 @@
+        require(
+            drips[_name].status == DripStatus.NONE,
+            "DrippieRedd: drip with that name already exists"
+        );
+
+        drips[_name] = DripState({ status: DripStatus.ACTIVE, config: _config, last: 0 });
+
+        emit DripCreated(_name, _config);
+    }
+
+    /**
+     * Configures a drip by name.
+     *
+     * @param _name Name of the drip to configure.
+     * @param _config Drip configuration.
+     */
+    function update(string memory _name, DripConfig memory _config) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        drips[_name].config = _config;
+
+        emit DripConfigUpdated(_name, _config);
+    }
+
+    /**
+     * Toggles the status of a given drip.
+     *
+     * @param _name Name of the drip to toggle.
+     */
+    function toggle(string memory _name) public onlyOwner {
+        require(
+            drips[_name].status != DripStatus.NONE,
+            "DrippieRedd: drip with that name does not exist"
+        );
+
+        if (drips[_name].status == DripStatus.ACTIVE) {
+            drips[_name].status = DripStatus.PAUSED;
+        } else {
+            drips[_name].status = DripStatus.ACTIVE;
+        }
+
+        emit DripStatusUpdated(_name, drips[_name].status);
+    }
+
+    /**
+     * Triggers a drip.
+     *
+     * @param _name Name of the drip to trigger.
+     */
+    function drip(string memory _name) public {
+        DripState storage state = drips[_name];
+
+        require(
+            state.status == DripStatus.ACTIVE,
+            "DrippieRedd: selected drip does not exist or is not currently active"
+        );
+
+        // Don't drip if the drip interval has not yet elapsed since the last time we dripped. This
+        // is a safety measure that prevents a malicious recipient from, e.g., spending all of
+        // their funds and repeatedly requesting new drips. Limits the potential impact of a
+        // compromised recipient to just a single drip interval, after which the drip can be paused
+        // by the owner address.
+        require(
+            state.last + state.config.interval <= block.timestamp,
+            "DrippieRedd: drip interval has not elapsed since last drip"
+        );
+
+        // Checkscript is a special system for allowing drips to execute arbitrary EVM bytecode to
+        // determine whether or not to execute the drip. A checkscript is a simply EVM bytecode
+        // snippet that operates with the following requirements:
+        // 1. Stack is initialized a single value, address of the drip recipient.
+        // 2. Script can do any logic it wants.
+        // 3. Script can signal a successful check by leaving a 1 at the top of the stack.
+        // 4. Any value other than a 1 on the stack will signal a failed check.
+        bytes memory checkscript = state.config.checkscript;
```

Comment:
Should it return true/false or the 32 byte value?

Reply:
Maybe `abi.encode(bool,bytes32)` ? It feels like it would be really nice to have to test out checkscripts as you develop. Could leave this out if you think that its not something that would be used

Reply:
Alright I'll meet you in the middle and do just (bool) for simplicity.